### ISSUE TITLE
CCScheduler and CCActionManager Refactoring

### ISSUE THREAD

**Issue by slembcke** [cocos2d/cocos2d-objc#478]
The action manager and scheduler are very closely related classes and there is some very similar code shared between the two (or at least there was). It probably makes sense to just merge the two into the same entity. This would simplify a lot of code.

The other thing to consider is that we need to make actions work well with fixed updates  and physics. Combining the scheduler and action manager will make this easier to do effectively since the fixed update and all information about it is encapsulated by the scheduler. I definitely feel like it would be easier to achieve better action/physics integration if the scheduler and action manager were combined.

**Comment by Birkemose** [comment]
I agree, that these should probably be one class. I suggest the user should be allowed to create his own instances, if he has special scheduler or action needs. When running ex. many actions, the user might not want to have them all in one big pile.

**Comment by slembcke** [comment]
This has been discussed again in relation to v4. I'm bumping it to 4.x.
